sdk/samples/iot/iot_sample_common.h

```diff
@@ -44,16 +44,10 @@
     (void)printf("\n");        \
   } while (0)
 
-#define IOT_SAMPLE_LOG_AZ_SPAN(span_description, span)                       \
-  do                                                                         \
-  {                                                                          \
-    (void)printf("\t\t%s ", span_description);                               \
-    char* buffer = (char*)az_span_ptr(span);                                 \
-    for (int32_t az_span_i = 0; az_span_i < az_span_size(span); az_span_i++) \
-    {                                                                        \
-      putchar(*buffer++);                                                    \
-    }                                                                        \
-    (void)printf("\n");                                                      \
+#define IOT_SAMPLE_LOG_AZ_SPAN(span_description, span)                                        \
+  do                                                                                          \
+  {                                                                                           \
+    (void)printf("\t\t%s %.*s\n", (span_description), az_span_size(span), az_span_ptr(span)); \

```

Comment:
@momuno - is this an acceptable change? I believe the output will be identical. Please verify.

Reply:
In the guidelines, I see:  https://azure.github.io/azure-sdk/clang_design.html#clang-design-naming-macros3

` do { `

on the same line.  Which do we want to go with?

Reply:
Oh, I see the bigger change now.
I recall asking the same question to @hihigupt , and there is a reason for using putchar instead of printf.  I believe it has to do with printing non ASCII characters.  Since we don't know what could come in the AZ_SPAN, we don't know if it is all ASCII.  I understand there to be unique scenarios where printf does not print non-ASCII  characters correctly.  @hihigupt can you please clarify?  Thanks.

Reply:
Yes, I remember now:
https://github.com/Azure/azure-sdk-for-c/pull/911#discussion_r452443598

If the `span` contained a `0` byte, then this `printf` won't print all the bytes and stop there.
The `putchar` in a loop approach makes sure all the bytes of the span are printed.

Reply:
That's it.  Thanks for remembering!  So, no please do not make that change :)

Reply:
Oh, I see. I replaced it with `fwrite()` that doesn't stop at 0 terminator.

Reply:
@momuno , @hihigupt , I replaced with `fwrite()` which doesn't stop when it reads 0, looks good?

Reply:
Seems reasonable. The only nit I have is that some IoT devices don't have a filesystem, so using `fwrite` in samples (which is primarily for files, even though it is writing to stdout), might give the wrong impression to customers.

Reply:
It is hard for to tell whether it will be missing, or just support `stdout` when it is, but I won't be surprised it `fwrite(..., stdout)` does work there, and then we are doing it effectively, by not outputting bytes one-by-one. IoT team, I leave this to you - do you have info whether IoT compilers support fwrite to stdout?

Reply:
Let's unblock and merge this PR in. If necessary, folks can follow up and revert.

Reply:
Hi.  I am adding @hihigupt to this.